tools/restatectl/src/commands/metadata/put.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::commands::metadata::patch::{patch_value, PatchValueOpts};
+use crate::commands::metadata::MetadataCommonOpts;
+use anyhow::anyhow;
+use clap::Parser;
+use clap_stdin::FileOrStdin;
+use cling::{Collect, Run};
+use serde_json::Value;
+use std::collections::HashMap;
+
+#[derive(Run, Parser, Collect, Clone, Debug)]
+#[clap()]
+#[cling(run = "put_value")]
+pub struct PutValueOpts {
+    #[clap(flatten)]
+    metadata: MetadataCommonOpts,
+
+    /// The key to patch
+    #[arg(short, long)]
+    key: String,
+
+    /// The JSON document to store, can be read from stdin or a file path
+    #[arg(short, long)]
+    content: FileOrStdin,
+
+    /// Expected version for conditional update
+    #[arg(short = 'e', long)]
+    version: Option<u32>,
+
+    /// Preview the change without applying it
+    #[arg(short = 'n', long, default_value_t = false)]
+    dry_run: bool,
+}
+
+async fn put_value(opts: &PutValueOpts) -> anyhow::Result<()> {
+    let opts = opts.clone();
+
+    let doc_body = opts.content.contents().map_err(|e| anyhow!(e))?;
+
+    let mut doc: Value = serde_json::from_str(&doc_body)
+        .map_err(|e| anyhow::anyhow!("Parsing JSON value: {}", e))?;
```

Comment:
```suggestion
    let mut doc: Value = serde_json::from_str(&doc_body).context("Failed to parse JSON value")?;
```
Although I don't think we need either solutions.